### PR TITLE
[SPARK-16130][ML] model loading backward compatibility for ml.classfication.LogisticRegression

### PR DESCRIPTION
**Pull request by hhbyyh**
## What changes were proposed in this pull request?

jira: https://issues.apache.org/jira/browse/SPARK-16130
model loading backward compatibility for ml.classfication.LogisticRegression
## How was this patch tested?

existing ut and manual test for loading old models.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala [hunk 1]
```diff
@@ -674,12 +674,13 @@ object LogisticRegressionModel extends MLReadable[LogisticRegressionModel] {
 
       val dataPath = new Path(path, "data").toString
       val data = sparkSession.read.format("parquet").load(dataPath)
-        .select("numClasses", "numFeatures", "intercept", "coefficients").head()
+        .select("numClasses", "numFeatures", "intercept", "coefficients")
+      val row = MLUtils.convertVectorColumnsToML(data, "coefficients").head()
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala]: I think we should call convert vector columns first and then call `select` to ensure the ordering of columns. There is no contact on keeping the column ordering between mappings.

Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala]: Btw, if we are going to update this PR, it is more readable to write

~~~scala
val Row(numClasses: Int, numFeatures: Int, intercept: Double, coefficients: Vector) = data.head()
~~~

**Comment by hhbyyh** [comment]
Thanks for the review @mengxr . Updated according to the comments.

File: mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala [hunk 2]
```diff
@@ -674,12 +674,12 @@ object LogisticRegressionModel extends MLReadable[LogisticRegressionModel] {
...
-        .select("numClasses", "numFeatures", "intercept", "coefficients").head()
+
       // We will need numClasses, numFeatures in the future for multinomial logreg support.
-      // val numClasses = data.getInt(0)
-      // val numFeatures = data.getInt(1)
-      val intercept = data.getDouble(2)
-      val coefficients = data.getAs[Vector](3)
+      val Row(numClasses: Int, numFeatures: Int, intercept: Double, coefficients: Vector) =
+        MLUtils.convertVectorColumnsToML(data, "coefficients")
+        .select("numClasses", "numFeatures", "intercept", "coefficients")
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala]: should have two more spaces

**Pull request closed without merging**